logicrunner/logicrunner_unit_test.go

```diff
@@ -150,42 +144,61 @@ func (suite *LogicRunnerTestSuite) TestStartQueueProcessorIfNeeded_DontStartQueu
 func (suite *LogicRunnerTestSuite) TestCheckPendingRequests() {
 	objectRef := testutils.RandomRef()
 
-	es := &ExecutionState{ArtifactManager: suite.am}
-	pending, err := es.CheckPendingRequests(
-		suite.ctx, &message.CallConstructor{},
-	)
-	suite.Require().NoError(err)
-	suite.Require().Equal(message.NotPending, pending)
+	suite.T().Run("already in pending", func(t *testing.T) {
+		parcel:= testutils.NewParcelMock(t)
+		es := &ExecutionState{Ref: objectRef, pending: message.InPending}
+		err := suite.lr.ClarifyPendingState(suite.ctx, es, parcel)
+		require.NoError(t, err)
+		require.Equal(t, message.InPending, es.pending)
+	})
 
-	suite.am.HasPendingRequestsMock.Return(false, nil)
-	es = &ExecutionState{ArtifactManager: suite.am}
-	pending, err = es.CheckPendingRequests(
-		suite.ctx, &message.CallMethod{
-			ObjectRef: objectRef,
-		},
-	)
-	suite.Require().NoError(err)
-	suite.Require().Equal(message.NotPending, pending)
+	suite.T().Run("already not in pending", func(t *testing.T) {
```

Comment:
it's better to use test tables, like in test TestLedgerHasMoreRequests

Reply:
done